conjur-api/ResourceKind.cs

```diff
@@ -0,0 +1,17 @@
+﻿namespace Conjur
+{
+    /// <summary>
+    /// Resource kind.
+    /// <see cref="https://www.conjur.org/api.html#header-kinds-of-resources"/>: Conjur API resource kinds
+    /// </summary>
+    public enum ResourceKind
```

Comment:
This is just wrong. Resource kind is a free form string.

Reply:
Can you provide a justification please. I think that enum really simplify it and made it less error prone.

Reply:
A common use case is when someone is going against our uniform structure - User vs user

Reply:
I'm not sure what you mean by 'justification'. Conjur API allows you to use any resource kind, which can be useful for custom solutions. It's a free-form field and not an enum; the catalog is open. The dotnet API library should not take away that ability.

If misspelling is a concern, what do you think about having some static string constants with the common kinds instead?

Reply:
I would like to understand it deeper. I've tried to upload a policy with custom resource name and received "Unrecognized data type" error. Is there public/privet API that allows customer to add new resource types (or it's in relatively close road map)?

Reply:
Another notice, this constructor is internal so anyway is limited to users.

Reply:
Here's an example of a policy with a custom kind:
```yaml
- !resource &north_gate
  kind: gate
  id: north

- !user bob

- !permit
  role: !user bob
  resource: *north_gate
  privilege: open
```

For this toy example, imagine you load this policy as root for account 'gates'. Now imagine you have a webservice where you can POST `open` to http://gates.example/north. The webservice could take Authorization header from the request (assuming it's Conjur bearer token) and use it to HEAD https://conjur.example/resources/gates/gate/north?check=true&privilege=open and only activate the gate if Conjur returns an ok response.

Reply:
And maybe the constructor is internal, but Client.Resource() method isn't.

Reply:
Thank you for the clarification - since we allow users to send public check request it do seem logical now to allow free-form strings.